lib/UI/appbar.dart

```diff
@@ -0,0 +1,121 @@
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+
+class MySliverAppBar extends SliverPersistentHeaderDelegate {
+
+  final int completedCount;
+  final VoidCallback onEyePressed;
+  var eyeIcon = Icons.visibility;
+
+  MySliverAppBar ({
+    required this.completedCount,
+    required this.onEyePressed,
+  });
+
+  @override
+  Widget build(
+      BuildContext context, double shrinkOffset, bool overlapsContent) {
+    double shift = min(1, shrinkOffset / (164 - 80));
+    return StackBar(completedCount: completedCount, onEyePressed: onEyePressed, shift: shift);
+  }
+
+  @override
+  double get maxExtent => 164;
+
+  @override
+  double get minExtent => 80;
```

Comment:
Твой аппбар при свёрнутом виде выглядит, будто слегка "урезанный" снизу. Чтобы лучше соответствовало макету, следует увеличить значение minExtent:
```suggestion
  double get minExtent => 90;
```